src/app.rs

```diff
@@ -0,0 +1,51 @@
+use crate::contract_leaderboard::get_contracts;
+use crate::db::get_db_pool;
+use crate::state::AppState;
+use axum::{routing::get, Router};
+use sqlx::PgPool;
+use std::sync::Arc;
+
+pub async fn get_app() -> Router {
+    let db_pool = get_db_pool().await;
+    get_app_with_db_pool(db_pool)
+}
+
+fn get_app_with_db_pool(db_pool: PgPool) -> Router {
+    let shared_state = Arc::new(AppState { db_pool });
+    tracing_subscriber::fmt::init();
+    Router::new()
+        .route("/contracts", get(get_contracts))
+        .with_state(shared_state)
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::contract_leaderboard::ContractEntry;
+    use axum::{body::Body, http::Request};
+    use tower::ServiceExt;
+
+    #[tokio::test]
+    async fn test_get_contracts_against_dev_db() {
+        let app = get_app().await;
+        let response = app
+            .oneshot(
+                Request::builder()
+                    .uri("/contracts")
+                    .body(Body::empty())
+                    .unwrap(),
+            )
+            .await
+            .expect("Request Failed");
+        assert_eq!(response.status(), 200);
+        let body = hyper::body::to_bytes(response.into_body()).await.unwrap();
+        let contracts: Vec<ContractEntry> = serde_json::from_slice(&body).unwrap();
+        assert_eq!(contracts.len(), 100);
+
+        for contract in contracts {
+            assert!(contract.address.len() > 0);
```

Comment:
Rust has an Option type. I'd prefer to use that. Let's agree to ban empty string. No empty string. Anywhere. Use `None` instead if you need to encode the absence of a string value. If you have encountered actual empty strings please let me know and I'll go fix it. Restricting the data is often even more powerful than restricting code functionality in trying to encode essential complexity and minimize accidental complexity. Empty strings are accidental imo. You never need them.

Reply:
This data comes straight from the db. I don't encode missing data anywhere but just use what is returned from the db.
So there should not be any empty strings here unless the db contains such data. (which I think is impossible since the address is the primary key in the contract_base_fees table).